TunnelVisionLabs.ReferenceAssemblyAnnotator/Program.cs

```diff
@@ -207,6 +220,25 @@ static bool IsMatchingConstructor(MethodDefinition constructor, CustomAttribute
             }
         }
 
+        private static bool IsExcludedAnnotation(ICustomAttributeProvider provider, ICustomAttributeProvider annotatedProvider, CustomAttribute customAttribute)
+        {
+            if (provider is ParameterDefinition parameter
+                && parameter.Method is MethodDefinition { Name: nameof(GetHashCode), Parameters: { Count: 1 } } method
+                && method.DeclaringType is { Namespace: "System.Collections.Generic", GenericParameters: { Count: 1 } } type)
+            {
+                if (type.Name == "IEqualityComparer`1"
+                    || type.Name == "EqualityComparer`1")
+                {
+                    // Remove DisallowNullAttribute from:
+                    // 1. System.Collections.Generic.IEqualityComparer<T>.GetHashCode([DisallowNull] T)
+                    // 2. System.Collections.Generic.EqualityComparer<T>.GetHashCode([DisallowNull] T)
```

Comment:
Why not just remove from \#2?

Reply:
(I typed `\#2`. It hid the backslash without escaping anything. Didn't that once escape # links?)

Reply:
The interface should not be preventing a user from implementing an equality comparer for a value domain where `null` is a valid value.

Reply:
Except it does prevent it. A type should be able to have `IEqualityComparer<MyObject?>` as an input, knowing that the type needs a comparer capable of operating on any `MyObject` instance or `null`, and it should not be required to be tightly coupled to one implementation of the interface in order to do so.

Reply:
Ahh. I see how it prevents consumption of `IEqualityComparer` while allowing null, not how it prevents implementation (per se) of `IEqualityComparer` while allowing null, but I think I'm following.